packages/gravity/blade-runner/src/plan/env/scheduler-env.ts

```diff
@@ -0,0 +1,201 @@
+//
+// Copyright 2024 DXOS.org
+//
+
+import { type Callback, Redis, type RedisOptions } from 'ioredis';
+import path from 'node:path';
+
+import { Trigger } from '@dxos/async';
+import { log } from '@dxos/log';
+
+import { ReplicantRpcHandle, open, close } from './replicant-rpc-handle';
+import { REDIS_PORT, createRedisRpcPort } from './util';
+import { WebSocketRedisProxy } from './websocket-redis-proxy';
+import { type Replicant, type ReplicantBrain, type SchedulerEnv, type RpcHandle } from '../interface';
+import { type ProcessHandle, runBrowser, runNode } from '../run-process';
+import {
+  type ReplicantRuntimeParams,
+  type ReplicantParams,
+  type GlobalOptions,
+  type TestParams,
+  AGENT_LOG_FILE,
+  type ReplicantsSummary,
+} from '../spec';
+
+export class SchedulerEnvImpl<S> implements SchedulerEnv<S> {
+  public redis: Redis;
+
+  // Redis client for subscribing to sync events.
+  public redisSub: Redis;
+
+  // Start websocket REDIS proxy for browser tests.
+  private readonly _server = new WebSocketRedisProxy();
```

Comment:
all can be readonly?
order variables by readonly, then ones set in constructor, then private, then public